### PR TITLE
Job pref revamp

### PR DESCRIPTION
**Pull request by AnturK**
Switches from flag-based system to simple keyed list.
Kills job flags and department flags.

Unfortunately i can't see sensible way to update prefs without writing godzilla size switch or keeping the unused variables.

### CODE REVIEW

**Comment by MrPerson** [comment]
Maybe keep the old vars but only for a month or so. It would really suck to make everyone have to redo their job prefs.

**Comment by zxaber** [comment]
Wouldn't be that big of a deal. Make super giant text in the welcome text for a few weeks and call it a day. Worst case, people get like one or two rounds in an unwanted job.

**Comment by SpaceManiac** [comment]
We have a whole system for savefile portability, it should be used.

Good change overall, I might be wary of making `job.title` a savefile-stable variable though.

**Comment by AnturK** [comment]
Yeah, i was considering adding separate job id instead, but then i'd probably have to think what to do with jobbans. I guess i'll just go with leaving the unused vars for a moment.

**Comment by AnturK** [comment]
Not sure if i should be implementing the job id, pretty much everything relies on titles being constant already. I'll just add some warnings to it for now.

File: code/modules/client/preferences_savefile.dm [hunk 1]
```diff
@@ -51,7 +51,60 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
...
+						new_value = JP_LOW
+				if(ENGSEC)
+					if(job_engsec_high & fval)
+						new_value = JP_HIGH
+					else if(job_engsec_med & fval)
+						new_value = JP_MEDIUM
+					else if(job_engsec_low & fval)
+						new_value = JP_LOW
+			if(new_value)
+				to_chat(world,"Updating [initial(J.title)] to [new_value]")
```
Comment (MrPerson) [inline code/modules/client/preferences_savefile.dm]: Just a reminder that this is here since I expect you'll want to delete this later

Comment (AnturK) [reply code/modules/client/preferences_savefile.dm]: Oh, whoops forgot about this.

**Comment by optimumtact** [comment]
is this still WIP or is it ready for review?

**Comment by AnturK** [comment]
It's ready

**Review by optimumtact — APPROVED** [review]
LGTM

**Pull request merged**